tests/jobs/test_scheduler_job.py

```diff
@@ -3776,6 +3776,11 @@ def test_create_dag_runs_datasets(self, session, dag_maker):
         dataset1 = Dataset(uri="ds1")
         dataset2 = Dataset(uri="ds2")
 
+        # Create DAG before the arrival of dataset events.
```

Comment:
Should this be part of its own, new test?

`test_new_dagrun_ignores_old_datasets` or similar?

Reply:
@RNHTTR I've added a new unit test. Please review it and let me know if it needs any adjustments. 

Unit test is checking newly added Dag should just get the Dataset arrived after it was created(event3) and not the dataset events(event 1 and event2)